dvc/remote/base.py

```diff
@@ -309,18 +322,23 @@ def changed(self, path_info, checksum_info):
             logger.debug("'{}' doesn't exist.".format(path_info))
             return True
 
-        checksum = checksum_info.get(self.PARAM_CHECKSUM)
+        checksum = None
```

Comment:
You could extract this logic into something like:
```
  def changed_checksum(self, path_info, checksum_info):                     
        if not checksum_info:                                                 
            return True                                                       
                                                                              
        types = set(checksum_info.keys())                                     
        supported_types = set(self.checksum_types)                            
        intersection = supported_types.intersection(types)                    
        if not supported_types.intersection(types):                                                          
            return True                                                       
                                                                              
        for typ in intersection:                                              
            expected = checksum_info.get(typ)                                      
            actual = self.get_checksum(path_info, typ)                        
            if actual != expected:
                return True                                                   
                                                                              
        return False                                                          
```

and use it both in remote.changed() and in output.changed_checksum()